code/Base/GamemodeSystem/GamemodeOrb.cs

```diff
@@ -4,31 +4,27 @@ namespace Sports;
 [Library( "sports_gamemode_orb" )]
 [Title( "Gamemode Orb" )]
 [Category( "Map Setup" )]
-[Particle( "orbparticlepath" )]
+[EditorModel( "models/editor/cone_helper.vmdl" )]
+[Particle]
 public partial class GamemodeOrb : BaseTrigger
 {
 	public static HashSet<GamemodeOrb> Orbs { get; set; } = new();
 
-	[Property, ResourceType( "vpcf" )]
-	private string orbParticlePath { get; set; } = "particles/gamemode_orb/orb.vpcf";
-	private ParticleSystemEntity orbParticle { get; set; }
-	[Property( "snapshot_file" ), ResourceType( "vsnap" )]
-	public string SnapshotFile { get; set; }
-	[Property( "snapshot_mesh" ), FGDType( "node_id" )]
-	private string SnapshotMesh { get; set; }
-
-	[Property]
-	public string GamemodeId { get; set; }
-
+	[Property( "effect_name", Title = "Particle System Name" ), ResourceType( "vpcf" )]
+	public string ParticleSystemName { get; set; }
 	public BaseGamemode LinkedGamemode { get; set; }
 
+	[Property( "gamemode_name", Title = "Gamemode Name" ), FGDType( "target_destination" )]
+	public string Gamemode { get; set; }
+
 	protected TimeSince LastTouch = 1f;
 
 	[Event.Entity.PostSpawn]
 	protected void PostEntitiesSpawned()
 	{
-		LinkedGamemode = SportsGame.Instance?.GetGamemodeFromId( GamemodeId );
-		Log.Debug( "Orb: Linked to gamemode." );
+		LinkedGamemode = SportsGame.Instance?.GetGamemodeFromId( Gamemode );
+		if ( LinkedGamemode is not null )
```

Comment:
I would prefer using IsValid for Entities. It's just something I've done everywhere else. What's your take on this?

Reply:
I'm fine with that, IsValid does a null check anyways, so it's probably safer to use in the sbox context